executors.go

```diff
@@ -10,6 +11,22 @@ import (
 
 // Reload fetch fresh data for a given model, using its ID
 func (c *Connection) Reload(model interface{}) error {
+	v := reflect.Indirect(reflect.ValueOf(model))
```

Comment:
I feel like a lot of this logic is repeated. would make sense to break it out into it's own method?